src/views/Nodes.vue

```diff
@@ -111,6 +120,11 @@ export default {
     sliceString(string, symbols) {
       return string.length > symbols ? `${string.slice(0, symbols)}…` : string;
     },
+    updateHideOfflineNodes() {
+      const hideOffline = !this.hideOfflineNodes ? 1 : undefined
```

Comment:
Very confused by this. If _not_ hideOfflineNodes, set hideOffline=1?

Reply:
@annybs - it's because this is the function to update hideOfflineNodes. If true, set to false, and vice versa.

So, when the user clicks on the checkbox:

- if `hideOfflineNodes` is true (`?hideOffline=1`),  then the function will set the `hideOffline` query to `undefined`. 
- if `hideOfflineNodes` is false (`?hideOffline` is anything but `1`), then the function will set `?hideOffline=1`

Reply:
OK. I think what threw me was that I misread the component structure as having a `hideOfflineNodes` _data_ state, and I wasn't seeing a change to the state in the form of `this.hideOfflineNodes = !this.hideOfflineNodes`

So what's actually happening here is a redirect, which causes the component _prop_ to change from the outside. It's still not über intuitive to me, but I can't fault the code. Approved 👍

Reply:
@annybs - exactly. As I can't maintain the _data_ state upon redirect (unless we use something like vuex), the state of `hideOfflineNodes` is determined as a computed prop from the presence of the url query `?hideOffline=1`. And then to update the state, I have to redirect rather than update state directly.